app/models/open_id_provider.rb

```diff
@@ -0,0 +1,6 @@
+class OpenIdProvider < ApplicationRecord
+  belongs_to :account
+
+  validates :sub, presence: true
+  validates :provider, presence: true
+end
```

Comment:
```suggestion
  validates :sub, uniqueness: { scope: :provider }
end
```
モデル側でも複数カラムでのユニーク成約を書く場合はこんな感じになりそうです。
この場合、`account.save` をする前に `valid?` で判定できるようになり、
DB アクセスの回数を減らせそうです。(実際にはかなりわずかな影響だと思いますが。。)

ですが、この変更をすると修正箇所がいくつか出てくるので、
他タスクを消化して余力があれば～で良いのかなという気もしています。

モデルでの複数カラムによるユニーク成約について、参考記事はこのあたりです。
https://guides.rubyonrails.org/active_record_validations.html#uniqueness

Reply:
ありがとうございます、モデルのバリデーションでユニーク制約できること知りませんでした！
こちら仰る通りテストにも影響するため修正箇所がここだけでなくなるので、一旦Issue切って、後で可能であれば対応～、という形にします。
https://github.com/commew/timelogger-api/issues/48

Reply:
issue 作成ありがとうございます！
対応方針について承知しました！